docs: Update introduction. (#125)
nodejs/nodejs.dev: src/documentation/0001-node-introduction/index.md

---
title: Introduction to Node.js
description: "Getting started guide to Node.js, the server-side JavaScript runtime environment. Node.js is built on top of the Google Chrome V8 JavaScript engine, and it's mainly used to create web servers - but it's not limited to just that."

> author: flaviocopes

Reviewer: The `authors` variable in the frontmatter is used to generate the `Contributors` section at the bottom of every page.

![image](https://user-images.githubusercontent.com/2608321/54472602-f3d42300-47f0-11e9-9ac0-97cb4c4b6840.png)

```suggestion
authors: flaviocopes
```

Reviewer 2: Done.

Not for this PR, but: I guess I sorta understand the reason for naming specific authors, but unless this document doesn't get updates and falls horribly out-of-date, that list is pretty much guaranteed to get ridiculously long or else to not actually reflect authorship of the material. I wonder if something else can be done to provide credit, such as a footer link to an AUTHORS file. (That too is likely to be out of date or ridiculously long, but at least it won't be an in-page distraction.)

Reviewer: Thanks for the input, @Trott! I absolutely agree with you. 

Will raise an issue to open discussion on this.

Reviewer: Discussion continued on https://github.com/nodejs/nodejs.dev/issues/201.

---
title: Introduction to Node.js
description: "Getting started guide to Node.js, the server-side JavaScript runtime environment. Node.js is built on top of the Google Chrome V8 JavaScript engine, and it's mainly used to create web servers - but it's not limited to just that."
author: flaviocopes
section: Quick Start
---

Node.js is a **runtime environment for JavaScript** that runs on the **server**.

> Node.js is open source and cross-platform. Since its introduction in 2009, it got hugely popular and now plays a significant role in the web development scene. If GitHub stars are one popularity indication factor, having 58,000+ stars means being extremely popular.

Reviewer: ```suggestion
Node.js is open source and cross-platform. Since its introduction in 2009, it got hugely popular and now plays a significant role in the web development scene. If GitHub stars are any indication of popularity, having 58,000+ stars means being extremely popular.
```

Reviewer 2: This text was eliminated by another commit. Resolving.

> Node.js runs the V8 JavaScript engine, the core of Google Chrome, outside of the browser. Node.js is able to leverage the work of the engineers that made (and will continue to make) the Chrome JavaScript runtime blazing fast, and this allows Node.js to benefit from the substantial performance improvements and the Just-In-Time compilation that V8 performs. Thanks to this, JavaScript code running in Node.js can become very performant.

Reviewer: ```suggestion
Node.js runs the V8 JavaScript engine, the core of Google Chrome, outside of the browser. Node.js is able to leverage the work of the engineers that made (and continue to make) the Chrome JavaScript runtime blazing fast, and this allows Node.js to benefit from the substantial performance improvements and the Just-In-Time compilation that V8 performs. Thanks to this, JavaScript code running in Node.js can become very performant.
```

> A Node.js app is run by a single process, without creating a new thread for every request. Node provides a set of asynchronous I/O primitives in its standard library that will prevent JavaScript code from blocking and generally, libraries in Node.js are written using non-blocking paradigms, making a blocking behavior an exception rather than the normal.

Reviewer: ```suggestion
A Node.js app is run in a single process, without creating a new thread for every request. Node.js provides a set of asynchronous I/O primitives in its standard library that prevent JavaScript code from blocking and generally, libraries in Node.js are written using non-blocking paradigms, making blocking behavior the exception rather than the norm.
```

> When Node.js needs to perform an I/O operation, like reading from the network, access a database or the filesystem, instead of blocking the thread Node.js will resume the operations when the response comes back, instead of wasting CPU cycles waiting.

Reviewer: ```suggestion
When Node.js needs to perform an I/O operation like reading from the network or accessing a database or the filesystem, instead of blocking the thread and wasting CPU cycles waiting, Node.js will resume the operations when the response comes back.
```

> This allows Node.js to handle thousands of concurrent connections with a single server without introducing the burden of managing threads concurrency, which would be a significant source of bugs.

Reviewer: ```suggestion
This allows Node.js to handle thousands of concurrent connections with a single server without introducing the burden of managing thread concurrency, which could be a significant source of bugs.
```

> Node.js has a unique advantage because millions of frontend developers that write JavaScript for the browser are now able to run the server-side code and frontend-side code without the need to learn a completely different language.

Reviewer: ```suggestion
Node.js has a unique advantage because millions of frontend developers that write JavaScript for the browser are now able to write the server-side code in addition to the client-side code without the need to learn a completely different language.
```

> In Node.js the new ECMAScript standards can be used without problems, as you don't have to wait for all your users to update their browsers - you are in charge of deciding which ECMAScript version to use by changing the Node.js version, and you can also enable specific experimental features by running Node with flags.

Reviewer: ```suggestion
In Node.js the new ECMAScript standards can be used without problems, as you don't have to wait for all your users to update their browsers - you are in charge of deciding which ECMAScript version to use by changing the Node.js version, and you can also enable specific experimental features by running Node.js with flags.
```
